### PR TITLE
[TASK-30] Show user name and image with sidebar specs.

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task Card - https://github.com/rafayet-monon/surveyor/projects/1#card-51278200

## What happened
See user name and profile photo
 
## Insight
**No user name provided in the API. Using type for now. User photo for dev is not available** 

- [x] Show user image and email
- [x] Write sidebar unit tests
- [x] Write sidebar UI tests

## Proof Of Work

- CI should pass successfully.

### CODE REVIEW

File: cypress/integration/sidebar.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const autoRecord = require('cypress-autorecord');
+
+describe('The Sidebar', () => {
+  autoRecord();
+
+  beforeEach(() => {
+    cy.login()
+  })
+
+  it('click in the user avatar opens sidebar', function () {
```
Comment (Lahphim) [inline cypress/integration/sidebar.spec.js]: @rafayet-monon The spec should describe about `sidebar` not about login and logout? 🤔
In my idea, it should be something like the sidebar open and close.
What do you think?
If we want to check about `login` `logout`, it should be in the another file

File: src/components/Sidebar/index.js [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+      await ProfileAdapter.get(authContext.state.authorization_token).then(
+        function (response) {
+          if (response.status === 200) {
+            setName(response.data.data.type);
+            setAvatar(response.data.data.attributes.avatar_url);
+          }
+        }
+      );
+    } catch (error) {
+      console.log(error);
```
Comment (Lahphim) [inline src/components/Sidebar/index.js]: @rafayet-monon What's the case that we get and error?
Is that the case about the the `authorization_token` is expired?

**Comment by rafayetn** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  
<b>🃏 FAIL</b> in <a href="https://github.com/rafayet-monon/surveyor/blob/feature/task-30-picture-and-email/src/tests/screens/Home/details.test.js">details.test.js</a>


<li>
<a href='https://github.com/rafayet-monon/surveyor/blob/feature/task-30-picture-and-email/src/tests/screens/Home/details.test.js#L45'>renders the user image of sidebar</a>
<br/>
Expected the element to have attribute: src="user-placeholder.jpg", received: null

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(element).toHaveAttribute("src", "user-placeholder.jpg") // element.getAttribute("src") === "user-placeholder.jpg"

Expected the element to have attribute:
  src="user-placeholder.jpg"
Received:
  null
    at Object.<anonymous> (/home/runner/work/surveyor/surveyor/src/tests/screens/Home/details.test.js:45:18)
    at Object.asyncJestTest (/home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at /home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:43:12
    at new Promise (<anonymous>)
    at mapper (/home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at /home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:73:41
</code></pre>
</details>
</li>
<br/>


  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>
<b>🃏 FAIL</b> in <a href="https://github.com/rafayet-monon/surveyor/blob/feature/task-30-picture-and-email/src/tests/components/Sidebar/index.test.js">index.test.js</a>


<li>
<a href='https://github.com/rafayet-monon/surveyor/blob/feature/task-30-picture-and-email/src/tests/components/Sidebar/index.test.js#L27'>shows user avatar</a>
<br/>
Expected the element to have attribute: src="https://api.adorable.io/avatar/dev@nimblehq.co", received: src="user-placeholder.jpg"

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(element).toHaveAttribute("src", "https://api.adorable.io/avatar/dev@nimblehq.co") // element.getAttribute("src") === "https://api.adorable.io/avatar/dev@nimblehq.co"

Expected the element to have attribute:
  src="https://api.adorable.io/avatar/dev@nimblehq.co"
Received:
  src="user-placeholder.jpg"
    at Object.<anonymous> (/home/runner/work/surveyor/surveyor/src/tests/components/Sidebar/index.test.js:27:23)
</code></pre>
</details>
</li>
<br/>



<li>
<a href='https://github.com/rafayet-monon/surveyor/blob/feature/task-30-picture-and-email/src/tests/components/Sidebar/index.test.js#L42'>shows user name and avatar in the menu</a>
<br/>
Expected the element to have attribute: src="https://api.adorable.io/avatar/dev@nimblehq.co", received: src="user-placeholder.jpg"

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(element).toHaveAttribute("src", "https://api.adorable.io/avatar/dev@nimblehq.co") // element.getAttribute("src") === "https://api.adorable.io/avatar/dev@nimblehq.co"

Expected the element to have attribute:
  src="https://api.adorable.io/avatar/dev@nimblehq.co"
Received:
  src="user-placeholder.jpg"
    at Object.<anonymous> (/home/runner/work/surveyor/surveyor/src/tests/components/Sidebar/index.test.js:42:22)
</code></pre>
</details>
</li>
<br/>

</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d486c2b6de7536acb4504c1ec821d752f8309699
</p>

File: cypress/integration/sidebar.spec.js (hunk 1 above)
Comment (rafayet-monon) [reply cypress/integration/sidebar.spec.js]: I agree, @Lahphim. Fixed in [fd49f0a](https://github.com/rafayet-monon/surveyor/pull/25/commits/fd49f0a34f10bd3490524fed5d51d1614938c7ff)

File: src/components/Sidebar/index.js (hunk 2 above)
Comment (rafayet-monon) [reply src/components/Sidebar/index.js]: @Lahphim Not really. If some error occurs in that specific API.

**Review by Lahphim — APPROVED** [review]

**Review by carryall — COMMENTED** [review]
@rafayet-monon Is there any way to update the user image on the server? because the current one is not a proper image URL and it doesn't show up 💭 
Screen Shot 2564-01-08 at 18 35 44

**Review by carryall — APPROVED** [review]

**Comment by rafayet-monon** [comment]
> @rafayet-monon Is there any way to update the user image on the server? because the current one is not a proper image > URL and it doesn't show up 💭 

@carryall I opened an issue in the backend Repo. I'm planning to fix it next week hopefully.

**Pull request merged**